site/cypress/e2e/Cypress-Gradeable/office_hours_queue.spec.js

```diff
@@ -0,0 +1,213 @@
+    cy.get('.delete_queue_btn').last().click();
+};
+const disableQueue = () => {
+    cy.logout();
+    cy.login();
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-course-settings').click();
+    cy.get('#queue-enabled').should('be.checked');
+    cy.get('#queue-enabled').check();
+};
+const openNewQueue = (queueName, queueCode='') => {
+    cy.get('#nav-sidebar-queue').click();
+    cy.get('#toggle_new_queue').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('#new_queue_code').type(queueName);
+    if (queueCode.length > 0) {
+        cy.get('#new-queue-token').type(queueCode);
+    }
+    else {
+        cy.get('#new_queue_rand_token').click();
+    }
+    cy.get('#open_new_queue_btn').click();
+};
+
+const changeQueueCode = (queueName, queueCode='') => {
+    cy.get('#toggle_filter_settings').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('#old_queue_code').select(queueName);
+    if (queueCode.length > 0) {
+        cy.get('#old_queue_token').type(queueCode);
+    }
+    else {
+        cy.get('#old_queue_rand_token').click(); //random
+    }
+    cy.get('#change_code_btn').click(); // update it
+};
+const expectedAlerts =  (success=0, error=0, success_text='', error_text='') => {
+    if (success > 0) {
+        cy.get('.alert-success').contains(success_text);
+    }
+    if (error > 0) {
+        cy.get('.alert-error').contains(error_text);
+    }
+};
+const switchUser = (account) => {
+    cy.logout();
+    cy.login(account);
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-queue').should('exist').click();
+};
+const studentJoinQueue = (queueName, queueCode, name) => {
+    cy.get('#name_box').clear();
+    cy.get('#name_box').type(name);
+    cy.get('#queue_code').select(queueName).invoke('val'); // in which queue you want to join
+    cy.get('#queue_code').should('contain', queueName);
+    cy.get('#token-box').type(queueCode);
+    cy.get('#join_queue_btn').should('exist').click();
+};
+const studentRemoveSelfFromQueue = () => {
+    cy.get('#leave_queue').should('exist').click();
+};
+const studentFinishHelpSelf = () => {
+    cy.get('#self_finish_help').should('exist').click();
+};
+const helpFirstStudent = () => {
+    cy.get('.help_btn').first().should('exist').click();
+};
+const finishHelpFirstStudent = () => {
+    cy.get('.finish_helping_btn').first().should('exist').click();
+};
+const removeFirstStudent = () => {
+    cy.get('.remove_from_queue_btn').first().should('exist').click();
+};
+const restoreFirstStudent = () => {
+    cy.get('.queue_restore_btn').first().should('exist').click();
+};
+const toggleFirstQueueFilter = () => {
+    cy.get('.filter-buttons').first().should('exist').click();
+};
+const closeFirstQueue = () => {
+    cy.get('.toggle-queue-checkbox').first().should('exist').click();
+};
+const emptyFirstQueue = () => {
+    cy.get('.empty_queue_btn').first().should('exist').click();
```

Comment:
I don't think that having all of these one line statements as separate functions is beneficial. Also `.should('exist').click();` isn't necessary, as the element has to exist in order to click it. `should('exist')` should be used when you want to assert the existence, and that is it.